manifests/config.pp

```diff
@@ -10,13 +10,72 @@
     system => true,
   }
 
+  case $facts['os']['family'] {
+    'RedHat':  {
+      $nologin_shell = '/sbin/nologin'
+      case $facts['os']['release']['major'] {
+        '7': {
+          systemd::unit_file { 'caddy.service':
+            path    => '/etc/systemd/system',
+            content => template('caddy/etc/systemd/system/caddy.service.erb'),
+            enable  => true,
```

Comment:
If you include this, it defines a service `caddy.service` meaning you'll have 2 service resources for the same thing. Probably not what you want.